include/zephyr/drivers/sensor/grow_r502a.h

```diff
@@ -66,6 +72,37 @@ enum r502a_led_color_idx {
 				(R502A_LED_CTRL(ctrl) | R502A_LED_COLOR(color) | \
 					R502A_LED_SPEED(speed) | R502A_LED_CYCLE(cycle))
 
+enum r502a_sec_level {
+	R502A_SEC_LEVEL_1 = 1,
+	R502A_SEC_LEVEL_2,
+	R502A_SEC_LEVEL_3,
+	R502A_SEC_LEVEL_4,
+	R502A_SEC_LEVEL_5
+};
+
+enum r502a_data_len {
+	R502A_PKG_LEN_32,
+	R502A_PKG_LEN_64,
+	R502A_PKG_LEN_128,
+	R502A_PKG_LEN_256
+};
+
+enum r502a_sys_param_set {
+	R502A_BAUD_RATE = 4,
+	R502A_SECURITY_LEVEL,
+	R502A_DATA_PKG_LEN
```

Comment:
if data_pkg_len is fixed using KCONFIG, why does driver allow to change using set param !!

Reply:
data_pkg_len can be read/write using SENSOR_ATTR_R502A_SYS_PARAM attribute. Default value of data_pkg_len is 128 if user needs to change this, then they need to change data_pkg_len using attr_set and then change the kconfig to modified len.

Reply:
what !! the purpose of Kconfig is to define static config which has to be set in device not the reverse..